lib/Runtime/Executor/ThreadPoolExecutor.cpp

```diff
@@ -218,6 +241,26 @@ ExecutionContext *ExecutionState::getRawResultContextPtr() const {
   return resultCtx_.get();
 }
 
+Placeholder *ExecutionState::createOrGetPlaceholder(llvm::StringRef name,
+                                                    TypeRef type) {
+  Placeholder *ph;
+
+  if (intermediatePlaceholders_.count(name)) {
+    // If the Placeholder already exists, return a pointer to it.
+    auto &storedPh = intermediatePlaceholders_.find(name)->second;
+    ph = storedPh.get();
+  } else {
+    // If the Placeholder does not exist, create one, remember it, and return a
+    // pointer to it.
+    auto newPh =
+        llvm::make_unique<Placeholder>(name, type, /*isTrainable=*/false);
```

Comment:
I also want to point out that this `type` is owned by the `SymbolTable`. Should I copy that into the `Executor` as well?

Reply:
You shouldn't have to, the symbol table is owned by the DAG node which will outlive the inference run.